packages/workbox-strategies/src/Strategy.ts

```diff
@@ -0,0 +1,236 @@
+/*
+  Copyright 2020 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {cacheNames} from 'workbox-core/_private/cacheNames.js';
+import {RouteHandlerObject, RouteHandlerCallbackOptions, WorkboxPlugin} from 'workbox-core/types.js';
+import {StrategyHandler} from './StrategyHandler.js';
+import './_version.js';
+
+
+export interface StrategyOptions {
+  cacheName?: string;
+  plugins?: WorkboxPlugin[];
+  fetchOptions?: RequestInit;
+  matchOptions?: CacheQueryOptions;
+}
+
+type StrategyHandlerOptions = {
+  request: Request;
+  event?: ExtendableEvent;
+  response?: Response;
+  params?: any;
```

Comment:
Should this be `string[] | {[paramName: string]: any}` to match the definition in `RouteHandlerCallbackOptions`? I'm  fine with making it looser in general, but I wonder if the mismatch could lead to some inadvertent type conflicts down the road.

Or I guess in general, is there a reason that the previously type types/interfaces aren't reusable here?

Reply:
I think I may have just overlooked this. As I was implementing the changes, I sometimes had to relax the types to just to get the code to compiles so I could experiment.

I'll revert this and use the same `MapLike` object type I'm going to add for `state`.